synapseclient/client.py

```diff
@@ -2098,15 +2098,21 @@ def getPermissions(self, entity, principalId=None):
                   or an empty array
 
         """
-        # TODO: what if user has permissions by membership in a group?
-        principalId = self._getUserbyPrincipalIdOrName(principalId)
+        principal_id = self._getUserbyPrincipalIdOrName(principalId)
         acl = self._getACL(entity)
+
+        team_list = self._find_teams_for_principal(principal_id)
+        team_ids = [int(team.id) for team in team_list]
+        effective_permission_set = set()
         for permissions in acl["resourceAccess"]:
-            if "principalId" in permissions and permissions["principalId"] == int(
-                principalId
+            if "principalId" in permissions and (
+                permissions["principalId"] == int(principal_id)
```

Comment:
```suggestion
                permissions["principalId"] == principal_id
```

nit.  IIRC, `_getUserbyPrincipalIdOrName()` returns an int, so this conversion may be redundant?  But then again, can't be too careful 🙃

Reply:
You are right - This type casting isn't required